### ISSUE TITLE
Active Scan Advanced Options Tab Consistency

### ISSUE THREAD

**Issue by kingthorin** [zaproxy/zaproxy#3538]
If you run an active scan and tweak the advanced options for input vectors, technology, etc during the setup of a "new" scan everything seems to stick the next time you start a new scan except the policy selection.

This seems to be related to:
`resetAndSetPolicy(String scanPolicyName)` in `CustomScanDialog`. I propose a change like:
```diff
        public void resetAndSetPolicy(String scanPolicyName) {
            logger.error("resetAndSetPolicy scanPolicyName: "+scanPolicyName);
            logger.error("resetAndSetPolicy scanPolicy.getName(): "+scanPolicy.getName());
-          policyAllCategoryPanel.reloadPolicies(scanPolicyName);
+          policyAllCategoryPanel.reloadPolicies(scanPolicy.getName());
        }
```

Does that make sense to everyone else? From my PoV if input vectors and technology selection, etc stick between custom scans I don't see any reason that policy shouldn't. Also having things sticky between scans seems handy from a use/re-use perspective.

The basic config (scope) tab will still default to the default policy (or whatever) but the policy selector on that tab is disabled while the advanced options are visible, so I think that's fine user/logic wise.

**Comment by thc202** [comment]
Makes sense to me.

**Comment by thc202** [comment]
The DB tab is not "reset" (to what was before) when the dialogue is cancelled but Input Vector tab is. Should we reset the tabs DB and Policy as well?

**Comment by kingthorin** [comment]
I'll get some screenshots. I think things should stick between runs.

**Comment by thc202** [comment]
Even when cancelled? (If started the options are kept in place.)

**Comment by kingthorin** [comment]
Can't do screenshots right now, but here's the flow. Start a new scan set input vectors, tech and policy ... run the scan. When that one is done (or maybe even while it runs...mine we're quick) Start another new scan, note tech and input vectors stuck from the previous scan, but policy didn't.

Cancel not implicated at all. However, I guess if you cancel it should go back to previous settings (not necessarily defaults).

**Comment by thc202** [comment]
Right, the expected behaviour is clear when the scan is started (keep the same options). Just wanted to know/clarify the expected behaviour when the dialogue is cancelled, which is reset to previous settings(?).

**Comment by kingthorin** [comment]
I think that makes the most sense.

**Comment by thc202** [comment]
OK, will tweak that too.

**Comment by kingthorin** [comment]
I understand there might be additional items along the same lines to be tweaked but can I go ahead and submit a PR for the original change? (It's really handy IMHO.)

**Comment by kingthorin** [comment]
@psiinon @thc202 good with the original change?

**Comment by thc202** [comment]
I'm fine with changing that but probably better to keep the issue open until everything is fixed.

The original change does not work correctly in all cases though, if you change the scan policy it goes back to current behaviour, it's better to move the statement `policyPanel.resetAndSetPolicy(scanPolicy.getName());` (from `init`) to the if statement in `reset`. The panel will behave more or less the same as is now but keeps the previous changes done to the scan policy (note that the policy might go stale, but there's always the reset button).

**Comment by davewichers** [comment]
OK. Given that it was decided that my issue #6096 is a duplicate of this issue, but this issue is still open after being created 3 years ago, when can someone make more progress on this? I'm mainly interested in having it remember customizations between scans in the same session. I don't actually know if you save a session and restore it whether those customizations carry over. That would be nice too, but isn't as important.

**Comment by kingthorin** [comment]
> when can someone make more progress on this?

When the person power or interest is sufficient for it to be tackled. As you know we're a very small core team of volunteers. While 3 years is a lot there are also other issues that are much older, however, that doesn't really figure into how things are tackled.
